### DISCUSSION TITLE
API Key doesn't allow for GPT help

### DISCUSSION THREAD

**Discussion by dgrolling** [plasma-umass/scalene#673] [Q&A] [answered]
I seem to have successfully installed scalene, but when I view the web report to review the code for optimization suggestions, I keep getting "# Query failed." messages anytime I click on the lightning bolt or the explosion icon. I have a valid OpenAI API key and have applied that in the advanced settings. Any ideas?

**Comment by emeryberger** [comment]
Thanks for the report. What system are you running on? If you can share a minimum working example of code where this happens, that might help.

**Reply by dgrolling** [reply]
Hello, this is running on a Windows VM. It fails on anything I try:

```def queryPORKADatabase(qry):
    import cx_Oracle
    #submit query
    connstr = 'XXXprd.cc.XX.XXX:1521/XXXprd_ha.cc.XX.XXX'
    conn = cx_Oracle.connect('XXXX', 'XXXX', connstr)
    curs = conn.cursor()
    try:
        curs.execute(qry)
    except:
        curs = ""
    return curs
    conn.close()
  ```

**Reply by emeryberger** [reply]
I'm still not sure why this wouldn't work, but I will push a branch that prints debug info to the JavaScript console so we can hopefully see what's going on.

**Reply by emeryberger** [reply]
Try this:

`python3 -m pip install git+https://github.com/plasma-umass/scalene@fixed_gpt_3_5_handling`

**Comment by London-Lass** [comment]
I have the same issue. I am using the personal api model at openapi and noticed that my free trial credits have expired. Does scalene rely on credits being available to make suggestions?

**Reply by emeryberger** [reply]
Yes, you definitely need credit available (even though it doesn't consume much). I will see what I can do to better surface this issue. @dgrolling is this possibly your issue too?

**Reply by emeryberger** [reply]
As a stopgap - if you could install the branch above and share the debugging message you see in the JavaScript console, that would be a great help. Thanks!

**Comment by London-Lass** [comment]
I topped up with a small amount of credit and works fine now.

**Comment by dgrolling** [comment]
Yes, that's my issue. Just wanted to test things out, but don't have any
credits.

**Comment by emeryberger** [comment, marked as answer]
The issue was insufficient funds. Thanks @dgrolling and @London-Lass ! I've updated the code (https://github.com/plasma-umass/scalene/pull/679) so there are now alerts for this situation as of the latest release (1.5.27).
